pyproject.toml

```diff
@@ -0,0 +1,101 @@
+[build-system]
```

Comment:
We should also update the install docs, in `doc/misc.rst`. Maybe also add a recommendation to *not* run `poetry` in an existing virtualenv. I may or may not have just been bitten by that. *cough*